app/controllers/content_items_controller.rb

```diff
@@ -147,11 +152,13 @@ def service_sign_in_format?(schema_name)
   def service_sign_in_presenter_name(content_item)
     slug = content_item_path.split("/").last
 
-    content_item["details"].each do |key, value|
-      if value["slug"] == slug
-        return "ServiceSignIn::#{key.classify}Presenter"
-      end
+    presenter_prefix = "choose_sign_in"
+
+    if content_item.dig("details", "create_new_account", "slug") == slug
+      presenter_prefix = "create_new_account"
     end
+
+    "ServiceSignIn::#{presenter_prefix.classify}Presenter"
```

Comment:
I don't think you need lines 155-158. You can just set line 161 to return the ChooseSignIn Presenter as the default:

`"ServiceSignIn::#ChooseSignInPresenter"`